### PR TITLE
dict2items filter

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

transforms a dict into a list of dicts with each element having a 'key' and 'value' key with the key and value of the original dict as values.

##### ISSUE TYPE

 - Feature Pull Request
##### COMPONENT NAME

filters/dict2items
##### ANSIBLE VERSION

```
2.5
```

### CODE REVIEW

File: lib/ansible/plugins/filter/core.py [hunk 1]
```diff
@@ -486,6 +486,14 @@ def flatten(mylist, levels=None):
     return ret
 
 
+def dict_to_list_of_dict_key_value_elements(mydict):
+    ''' takes a dictionary and transforms it into a list of dictionaries,
+        with each having a 'key' and 'value' keys that correspond to the keys and values of the original '''
+    ret = []
```
Comment (sivel) [inline lib/ansible/plugins/filter/core.py]: Should we ensure that `mydict` is actually a dict?

Could also "simplify" this with a list comprehension:

```
return [{'key': k, 'value': v} for k, v in mydict.items()]
```

Comment (bcoca) [reply lib/ansible/plugins/filter/core.py]: just copied the existing code, in any case that is 'shorten' not everyone thinks they are 'simpler'

**Comment by willthames** [comment]
Would be good to update the filters page here too.

**Comment by willthames** [comment]
rebuild_merge

I have plans to update the loops docs to reference this so I'll do the filters docs at the same time.

**Pull request merged**